### ISSUE TITLE
Bug: "Parsing error" for files that have been ignored

### ISSUE THREAD

**Issue by doberkofler** [typescript-eslint/typescript-eslint#8563]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-eslint

### Repro Code

```TypeScript
It seems as if typescript-eslint would wrongly or not at all skip the files defined in the `ignores` property.
This might be related to #8511 but does not only happen when `extends` is used.
```

### ESLint Config

```javascript
// @ts-check

import globals from 'globals';
import tseslint from 'typescript-eslint';

export default tseslint.config({
	ignores: [
		'vendor/**/*',
	],
	languageOptions: {
		ecmaVersion: 2021,
		sourceType: 'module',
		globals: {
			...globals.es2021,
		}
	},
	rules: {
		'indent': ['warn', 'tab'],
		'linebreak-style': ['warn', 'unix'],
		'quotes': ['warn', 'single'],
		'semi': ['warn', 'always'],
	},
},
{
	files: ['**/*.js'],
},
{
	files: ['**/*.ts'],
	plugins: {
		'@typescript-eslint': tseslint.plugin,
	},
	languageOptions: {
		parser: tseslint.parser,
		parserOptions: {
			project: true,
		},
	},
	rules: {
		'indent': 'off',
		'@typescript-eslint/indent': ['warn', 'tab'],
		'quotes': 'off',
		'@typescript-eslint/quotes': ['warn', 'single'],
		'semi': 'off',
		'@typescript-eslint/semi': ['warn', 'always'],
	},
});
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es2021",
    "module": "commonjs",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "skipLibCheck": true,
    "allowJs": true,
    "checkJs": true
  },
  "include": ["src/**/*.ts"]
}
```

### Expected Result

eslint should ignore the files in the `vendor` directory

### Actual Result

eslint reports parse errors for files in the directory `vendor`:

```bash
0:0  error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/vendor/file.d.ts` using `parserOptions.project`: tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `7.1.0` |
| `@typescript-eslint/parser`            | `7.1.0` |
| `TypeScript`                           | `5.3.3` |
| `ESLint`                               | `8.57.0` |
| `node`                                 | `20.11.0` |

**Comment by bradzacher** [comment]
you can use the `--print-config` flag to show the config for the file.
I suspect you're butting up against how flat configs work compared to before.

when you specify `ignores` **_it specifically only applies to the flat config array item that it is attached to_**.
Or to illustrate:
```ts
export default [
  // this config object will NEVER be applied to `src/ignored.ts`
  { ignores: ['src/ignored.ts'], ...someConfig1 },

  // this config object will ALWAYS be applied to `src/ignored.ts`
  { ...someConfig2 },

  // this config object will NEVER be applied to `src/ignored.ts` -- but only because it doesn't match the pattern
  { files: ['**/*.js'], ...someConfig3 },

  // this config object will ALWAYS be applied to `src/ignored.ts` -- but only because it doesn't match the pattern
  { files: ['**/*.ts'], ...someConfig4 },
];
```

It's a bit of a confusing switch from "legacy" configs where `ignorePatterns` applied to the entire config.

In flat configs if you want to specify global ignores you need to pass a config object with **JUST** the `ignores` property: https://eslint.org/docs/latest/use/configure/configuration-files-new#globally-ignoring-files-with-ignores
```ts
export default [
  // declares that src/ignored.ts is completely ignored by ESLint
  { ignores: ['src/ignored.ts'] },

  // this config object will NEVER be applied to `src/ignored.ts`
  { ...someConfig1 },

  // this config object will NEVER be applied to `src/ignored.ts`
  { ...someConfig2 },

  // this config object will NEVER be applied to `src/ignored.ts`
  { files: ['**/*.js'], ...someConfig3 },

  // this config object will NEVER be applied to `src/ignored.ts`
  { files: ['**/*.ts'], ...someConfig4 },
];
```

We _hint_ at this new behaviour in our `typescript-eslint` package docs: https://typescript-eslint.io/packages/typescript-eslint#usage-with-other-plugins

**Comment by doberkofler** [comment]
Thank you for the thoughtful explanation and sorry i did not spot this myself!

**Comment by bradzacher** [comment]
It's easy to miss - I messed up a few things and wasted an hour or so the first time I used flat configs. There are some subtle changes to be sure!
